Lib/pdb.py

```diff
@@ -159,11 +159,11 @@ def __init__(self, completekey='tab', stdin=None, stdout=None, skip=None,
         self.allow_kbdint = False
         self.nosigint = nosigint
 
-        # Read $HOME/.pdbrc and ./.pdbrc
+        # Read ~/.pdbrc and ./.pdbrc
         self.rcLines = []
         if readrc:
-            if 'HOME' in os.environ:
-                envHome = os.environ['HOME']
+            envHome = os.path.expanduser('~')
+            if envHome != '~':
```

Comment:
why do you test with '~' ? normally, you won't get the tilde but the full path. I have an other PR but you can check with my code and fix yours. https://github.com/python/cpython/pull/11855

Reply:
`envHome != '~'` is true if `expanduser` succeeded in finding the full path.

Reply:
yep, but in which case can you have '~' ? never

Reply:
Your solution is cleaner. We took the approach of modifying the fewest lines.

Reply:
in fact, expanduser will replace `~` by the $HOME, $HOMEDRIVE, $HOMEPATH, etc... for Windows and unix-like, so normally, you won't get the `~`. my approach is different because I wait for an exception and sometimes, an exception is better than a condition.

Reply:
You _could_ have `~` if the expansion fails (e.g. if it can't find anything resembling a home directory):  ["If the expansion fails or if the path does not begin with a tilde, the path is returned unchanged."](https://docs.python.org/2/library/os.path.html#os.path.expanduser)

Reply:
yep, it's possible.

1. use python3 for the documentation (https://docs.python.org/3/library/os.path.html#os.path.expanduser)
2. if you can't read the file you will get an exception, in this case, you will get OSError or Exception.

Reply:
Updated to use your cleaner solution